### PR TITLE
ale_linters: add 'dafny' linter

### PR DESCRIPTION
**Pull request by ttaylorr**
Here is a patch to add a linter for the Dafny programming language (https://rise4fun.org/Dafny).

This is my first contribution to ALE, so any feedback/criticism is appreciated 👍.

### CODE REVIEW

File: ale_linters/dafny/dafny.vim [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        \    'type': l:match[4] =~# '^Error' ? 'E' : 'W'
+        \ })
+    endfor
+    return l:output
+endfunction
+
+call ale#linter#Define('dafny', {
+\    'name': 'dafny',
+\    'executable': 'dafny',
+\    'command': 'dafny %s /compile:0',
```
Comment (w0rp) [inline ale_linters/dafny/dafny.vim]: Does this look at the file on disk, or does it read from stdin? If it's the former, add `'lint_file': 1`, so you can only check for problems on enter or on save.

Comment (ttaylorr) [reply ale_linters/dafny/dafny.vim]: @w0rp the `dafny` executable only reads files from disk, and I have updated this definition to match in da80129.

Comment (w0rp) [reply ale_linters/dafny/dafny.vim]: Okay, cool.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**